### ISSUE TITLE
Blazor async based polling with a timer

### ISSUE THREAD

**Issue by JonasZ95** [aspnet/AspNetCore#15151]
### Describe the solution you'd like
So I have the following service interface:
`
interface NewsService
{
 Task<NewsUpdate> GetNews(DateTime lastUpdate);
}
`
NewsUpdate contains a timestamp and an array of new News to be shown. I want to poll this service frequently for updates with a timer. This works as long you don't do It asynchronously, however System.Threading.Timer doesn't support async/await. So I would like some Timer which support async/await properly.

### Additional context
Using async void can cause unwanted behavior because the timer would not wait until one pass finished but instead spawn multiple tasks which can result in concurrency issues.

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @JonasZ95.
We have a sample showing how to do something similar here: https://docs.microsoft.com/en-us/aspnet/core/blazor/components?view=aspnetcore-3.0#invoke-component-methods-externally-to-update-state

**Comment by JonasZ95** [comment]
@mkArtakMSFT I don't see how this can solve the concurrency problems which is caused by the nature of the state machines which are generated for async functions.

Just look at this example:
```@page "/counter"
@using System.Threading
@implements IDisposable

<h1>Counter</h1>

<p>Current count: @_currentCount</p>

@code {

    Timer _updateTimer;
    int _currentCount = 0;
    
    public void Dispose()
    {
        _updateTimer.Dispose();
    }
    
    protected override async Task OnInitializedAsync()
    {
        _updateTimer = new Timer(state => { InvokeAsync(UpdateCounter); }, null, 0, 100);
    }

    async Task UpdateCounter()
    {
        _currentCount = 0;
        await Task.Delay(500);
        _currentCount++;
        await Task.Delay(500);
        _currentCount++;
        StateHasChanged();
    }
}
```

**Comment by NTaylorMullen** [comment]
> The key problem is that a new task is created while the previous task is stil running.

Just because it's created doesn't mean it has run. Calling InvokeAsync essentially queue's a task to be run on a specific thread, in this case Blazor's rendering thread. Meaning, any render operation will occur in the order in which it is enqueued. 

If you're worried about not awaiting your `InvokeAsync` you can mark your timer method as `async` (yes it'll be async void); events marked as async void are a supported pattern. That being said you'll still have to worry about your timer firing twice in parallel. That can be protected against in many different ways 😄

**Comment by JonasZ95** [comment]
@NTaylorMullen thanks for clarifying for the way how InvokeAsync works internally sadly there was close to no information regarding this. However, I already assumed that a Queue was used like this and I created the example under this assumption. Actually I think we don't need to discuss about `InvokeAsync` vs `async void` as long as the task plans to update the state.

Could you show one of the 'many different ways' to protect the timer firing twice? I tried this and I think It's kinda difficult to do this correctly.  Also, using the term 'parallel' is completely wrong here.
